From 77d55c4707395b76e32e07bf0f679b5b6999d568 Mon Sep 17 00:00:00 2001
From: Kai Kang <kai.kang@windriver.com>
Date: Thu, 27 Mar 2025 11:43:40 +0800
Subject: [PATCH] Fix initializer element is not constant

Fix error when compile for powerpc:

| x_sh_dbIO.c: In function 'swap_short':
| x_sh_dbIO.c:229:36: error: initializer element is not constant
|   229 |       static unsigned short ooop = *iptr;
|       |                                    ^

Upstream-Status: Pending

Signed-off-by: Kai Kang <kai.kang@windriver.com>
---
 src/sh_dbIO.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/sh_dbIO.c b/src/sh_dbIO.c
index 1f9b152..503456d 100644
--- a/src/sh_dbIO.c
+++ b/src/sh_dbIO.c
@@ -226,7 +226,8 @@ static unsigned short *  swap_short (unsigned short * iptr)
   else
     {
       /* alignment problem */
-      static unsigned short ooop = *iptr;
+      static unsigned short ooop;
+      ooop = *iptr;
       unsigned short hi   = (ooop & 0xff00);
       unsigned short lo   = (ooop & 0xff);
       ooop = (lo << 8) | (hi >> 8);
-- 
2.34.1

